vehicles/src/modules/permit/application.controller.ts

```diff
@@ -160,23 +161,39 @@ export class ApplicationController {
     @Param('permitId') permitId: string,
     @Query('amendment') amendment?: boolean,
   ): Promise<ReadApplicationDto> {
+    // Extracts the user object from the request, casting it to the expected IUserJWT type
     const currentUser = request.user as IUserJWT;
 
+    // Based on the amendment query parameter, selects the appropriate method to retrieve
+    // either the application or its current amendment, passing the permitId and current user for authorization and filtering
     const retApplicationDto = !amendment
       ? await this.applicationService.findApplication(permitId, currentUser)
       : await this.applicationService.findCurrentAmendmentApplication(
           permitId,
           currentUser,
         );
+
+    // Verifies user authorization to access the application data:
+    // Throws a ForbiddenException if the user does not have the correct authorization group or if the companyId from
+    // the application does not match the user's companyId. Additionally, checks if a company administrator is trying
+    // to access an application not originated online, which is not allowed.
     if (
-      currentUser.orbcUserAuthGroup === UserAuthGroup.COMPANY_ADMINISTRATOR &&
-      retApplicationDto.permitApplicationOrigin !==
-        PermitApplicationOrigin.ONLINE
+      (!doesUserHaveAuthGroup(
+        currentUser.orbcUserAuthGroup,
+        IDIR_USER_AUTH_GROUP_LIST,
+      ) &&
+        retApplicationDto?.companyId != currentUser.companyId) ||
+      (currentUser.orbcUserAuthGroup ===
+        ClientUserAuthGroup.COMPANY_ADMINISTRATOR &&
```

Comment:
```suggestion
      
```
Checking for Company admin is redundant and potentially leaves a security hole where a permit applicant may bypass the `PermitApplicationOrigin.ONLINE` check since they fail the company admin check.